lib/etl/tree.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Etl.Tree do
```

Comment:
this looks cool but would it be valuable (and easier to test) if you built the tree into a series of maps that could be json-encoded?

Reply:
probably, I included it in the PR, but it is by no means good.  I like the idea of a utility to visualize the stage tree.